enumerables.rb

```diff
@@ -0,0 +1,117 @@
+module Enumerable
+  def my_each()
+    return to_enum(:my_each) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i])
+    end
+    self
+  end
+
+  def my_each_with_index()
+    return to_enum(:my_each_with_index) unless block_given?
+
+    i = -1
+    arr = to_a
+    arr.length.times do
+      i += 1
+      yield(arr[i], i)
+    end
+    self
+  end
+
+  def my_select()
+    return to_enum(:my_select) unless block_given?
+
+    new_array = []
+    my_each do |item|
+      new_array << item if yield(item) == true
+    end
+    new_array.empty? ? self : new_array
+  end
+
+  def my_all?(pattern = nil)
+    my_each do |item|
+      if block_given?
+        return false unless yield(item)
+      elsif pattern.nil?
+        return false unless item
+      else
+        return false unless pattern === item
+      end
+    end
+    true
+  end
+
+  def my_any?(pattern = nil, &block)
+    my_each do |item|
+      if block_given?
+        return true unless block.call(item) == false
+      elsif pattern.nil?
+        return true unless item == false || item.nil?
+      else
+        return true unless (pattern === item) == false
+      end
+    end
+    false
+  end
+
+  def my_none?(pattern = nil, &block)
+    !my_any?(pattern, &block)
+  end
+
+  def my_count(num = 0, &block)
+    result = 0
+    my_each do |item|
+      if block_given? == false and item == num
+        result += 1
+      elsif block_given? and block.call(item) == true
+        result += 1
+      elsif num.zero? and block_given? == false
+        result += 1
+      end
+    end
+    result != 0 ? result : my_each { |x| x }
+  end
```

Comment:
- [ ] There is a bug in `my_count` method. When 0 is passed as an argument your method does not return the number of 0s inside the enumerator. It returns the length of the array instead. This bug happens only with 0. If another number is passed as an argument, the outcome is the expected one.

- Your code output =>  `p [0,1,2,3,4,5,5,5,7,8,9].my_count(0)` returns 11
- Expected output =>  `p [0,1,2,3,4,5,5,5,7,8,9].count(0)` returns 1